content/articles/how-to-implement-caching-in-adonisjs-5/index.md

```diff
@@ -0,0 +1,221 @@
+# How to Implement caching in Adonis.js 5.
+The performance of a web application is an integral part of the application. Many statistics have shown that slow web applications can result in loss of traffic, sales, and incomes in many different ways.
+
+In this article, we will learn how to implement a caching system on our Forum API developed with Adonis. We will also integrate the newly created caching package for the new Adonis 5 with optimum caching experience.
+
+In this article, we will continue and improve the performance of the Forum API developed on the [building Restful API with AdonisJS 5 tutorial](/engineering-education/build-a-restful-api-with-adonisjs/). You can take a peek at what we built in the previous article to know what the API is about.
+
+### Prerequisites
+You should have the following before proceeding with this tutorial.
+
+- Some understanding of TypeScript and Node.js.
+- [Some understanding of Adonis](https://masteringbackend.com/posts/adonisjs-tutorial-the-ultimate-guide)
+- [Some knowledge on building APIs with AdonisJS](/engineering-education/build-a-restful-api-with-adonisjs/)
+- [Some knowledge on caching](https://masteringbackend.com/posts/caching-strategy-for-restful-api/)
+
+### The goals of the tutorial
+I will be showing you the best practices for handling caching using Adonis. We will build a real-world forum app to demonstrate how to boost performance with a good caching strategy.
+
+In this tutorial, you will learn:
+- how to build a high-performance API with Adonis
+- how to implement a caching system in Adonis restful API.
+- about the concept of caching and how to scale large App with it.
+- the best practices for building performance-oriented web apps with Adonis.
+
+### Installling Adonis
+Getting started with Adonis 5 is very easy and straightforward as it is well-documented.
+
+If you're just getting started with Adonis 5, I recommend reading [AdonisJS 5 tutorial: The Ultimate Guide](https://masteringbackend.com/posts/adonisjs-tutorial-the-ultimate-guide) and [building a restful API with Adonis 5](/engineering-education/build-a-restful-api-with-adonisjs/) to get to the speed.
+
+You can run this command to have Adonis 5 installed.
+
+```bash
+npm init adonis-ts-app@latest forum-api
+```
+
+We will continue with the Forum API we have already developed with the article above. you can [clone the repository](https://github.com/Kaperskyguru/adonisjs-forum-api) to get started.
+
+### Setting up Adonis Cache
+After installing and setting up your Adonis 5 RestFul API project, we will install and configure our caching package. This will simplify implementing our caching strategy to the Forum API, to improve the performance or response time of the API.
+
+Run the following command to install the Adonis cache package.
+
+```bash
+npm i @kaperskyguru/adonis-cache
+```
```

Comment:
Tell us more about this package: `@kaperskyguru/adonis-cache`.

Reply:
Done